### DISCUSSION TITLE
scrape topic metrics from emqx broker

### DISCUSSION THREAD

**Discussion by chilphil** [emqx/emqx#5866] [General]
### Subject of the support
Hi,
I would like to monitor the consumer lagg of individual topics in my emqx broker via prometheus. I saw in your documentation that there are topic metrics that map this, right?(https://docs.emqx.io/en/enterprise/v4.3/modules/topic_metrics.html#create-module)

I would like to query these metrics from my prometheus instance. However, as far as I can see in Prometheus, these topic metrics do not appear there. Is there a way to get them? Or is there another way to determine the consumer lagg of each topic in prometheus? 

Greetings

### environment

- EMQ X version: 4.3.7:
- OS: Ubuntu 18.04.1 LTS
- Kernel : 4.15.0-36-generic

**Comment by qzhuyan** [comment]
@PA-Devol the link you refer to is enterprise, for open source version, you could config emqx to push metrics to`pushgateway` and Prometheus could pull from there.


pls refer to https://www.emqx.com/en/blog/emqx-prometheus-grafana
